pennylane/ops/functions/equal.py

```diff
@@ -15,44 +15,51 @@
 This module contains the qml.equal function.
 """
 # pylint: disable=too-many-arguments,too-many-return-statements
+from typing import Union
 import pennylane as qml
+from pennylane.measurements import MeasurementProcess
 from pennylane.operation import Operator
 
 
 def equal(
-    op1: Operator,
-    op2: Operator,
+    op1: Union[Operator, MeasurementProcess],
```

Comment:
Do I need to extend this functionality to `ShadowMeasurementProcess` and `MeasurementValue`? These are both classes in `measurements.py` which are related to `classical_shadow`, `shadow_expval`, and `qml.measure`

Reply:
It would be nice to add logic for the `ShadowMeasurementProcess` class: add comparison of `H` and `k` attributes.

Reply:
@albi3ro what about the `MeasurementValue` class?